go/apis/kube-mail/v1alpha1/smtpserver_types.go

```diff
@@ -0,0 +1,39 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type SMTPServerSpec struct {
+	Server   string `json:"server"`
+	Port     int    `json:"port"`
+	TLS      bool   `json:"tls"`
+	AuthType string `json:"authType"`
```

Comment:
All these are optional in the original spec:

https://github.com/mittwald/kube-mail/blob/6f9d3d3978be3e309a13edb60493ab2383594f07/src/k8s/types/v1alpha1/smtpserver.ts#L3-L8

Reply:
I've adjusted the struct tags and added enum validatio for `authType`.